### PR TITLE
Resampling functionality for NeuroVault

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
This PR updates what have been done in #1557 by @nicofarr and related to issue #1553

What have been done here in addition to prior work:

- rebase on master and resolve conflicts
- modify the neurovault mock to allow a fake image download (discussed with @jeromedockes in #1557)
- get the test up to date with the neurovault mock and split it into multiple ones

### CODE REVIEW

**Comment by RaphaelMeudec** [comment]
Codecov seems to fail because it compares test coverage right before the first commit of the PR (which was [last year](https://github.com/nilearn/nilearn/commit/4a3e943)). Though, I don't think there is a coverage problem with this PR as the functionality is tested.

**Review by bthirion — COMMENTED** [review]
I don't  see any outstanding issue here.
Simply add a whatsnew entry.
Best,

**Review by NicolasGensollen — APPROVED** [review]
This looks good to me. Thanks for finishing this PR @RaphaelMeudec ! :+1:

**Comment by jdkent** [comment]
I don't have any formal proof yet, but I believe the default interpolation strategy for resample_img (`continuous`) is sub-optimal for statistical maps. I was working on neurostuff/NiMARE#439, and found when running a meta-analysis using the `resample_img` defaults the results looked very pixelated and not biologically plausible, but when I used `linear` interpolation the results looked more plausible. Also changing `clip=True` may be worth consideration as to not introduce new larger statistical values than were in the original image.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -41,6 +41,10 @@ Enhancements
   colorbar of surface plots. The default format is scientific notation except for :func:`nilearn.plotting.plot_surf_roi`
   for which it is set as integers.
 
+- :func:`nilearn.datasets.neurovault.fetch_neurovault` now implements a `resample` boolean argument to either
```
Comment (NicolasGensollen) [inline doc/whats_new.rst:44]: I think you need to change the reference to `nilearn.datasets.fetch_neurovault`

**Comment by NicolasGensollen** [comment]
@jdkent do you want to open a separate issue on this? Also, do you have an example we could take a look at?

**Comment by bthirion** [comment]
@jdkent Thx for the comment
I'm just wondering whether we should acknowledge that not all images in Neurovault are statistical images: some of them are atlases, or at least display discrete labels. interpolation='linear' would not given sensible results for such images.
Add another keyword argument then ?

**Comment by RaphaelMeudec** [comment]
@NicolasGensollen I fixed the whatsnew entry. Not sure if I should proceed with the `interpolation` argument?

**Comment by bthirion** [comment]
I think we have to add this as another kw argument.

File: nilearn/datasets/neurovault.py [hunk 2]
```diff
@@ -2056,6 +2117,7 @@ def _read_download_params(
     wanted_image_ids=None, max_images=None,
     max_consecutive_fails=_MAX_CONSECUTIVE_FAILS,
     max_fails_in_collection=_MAX_FAILS_IN_COLLECTION,
+    resample=False, interpolation='continuous',
```
Comment (bthirion) [inline nilearn/datasets/neurovault.py:2120]: Following @jdkent comment, it would be better to use `interpolation=linear` as a default.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks!

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by NicolasGensollen** [comment]
Thank you all! :+1: 
Merging...

**Pull request merged**